app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,22 @@
+class FavoritesController < ApplicationController
+  before_filter :authenticate_user!
+  before_action :set_post
+
+  def create
+    favorite = Favorite.new(user_id: current_user.id, post_id: @post.id)
```

Comment:
You don't actually need to call the instance variable of `@post`. You can just call `post`.